app-backend/batch/src/main/scala/com/azavea/rf/batch/stac/ReadStacFeature.scala

```diff
@@ -1,11 +1,236 @@
+    }
+  }
+
+  protected def stacFeatureToScene(
+    feature: stac.Feature,
+    datasource: UUID,
+    params: CommandLine.Params
+  ): Scene.Create = {
+    val thumbnailLinks = feature.links.filter(_.`type` == "thumbnail")
+
+
+    // if datasource is defined, use bands from datasource
+    val (imageAssets, metadataFiles) = feature.assets.partition(asset => asset.format.getOrElse("none") == "tif")
+
+    val sceneId = UUID.randomUUID()
+    val images = getBandedImages(imageAssets, sceneId) // get bands from the image products
+
+    val srid = feature.geometry.srid
+
+    val geom = feature.geometry.geom.reproject(CRS.fromEpsgCode(srid), CRS.fromEpsgCode(3857))
+    val dataFootprint = geom.as[Polygon].map(g => Projected(MultiPolygon(g), 3857)).orElse(geom.as[MultiPolygon].map(g => Projected(g, 3857)))
+
+    Scene.Create(
+      id = Some(sceneId),
+      organizationId = landsat8Config.organizationUUID, // this config should be changed to a generic import config
+      ingestSizeBytes = 0,
+      visibility = Visibility.Public,
+      tags = feature.properties.provider.split(",").map(_.trim).toList,
+      datasource = datasource,
+      sceneMetadata = Json.Null, // stac feature fields that don't fit on the scene
+      name = feature.id,
+      owner = Some(systemUser),
+      tileFootprint = Some(multipolygonFromBbox(feature.bbox)),
+      dataFootprint = dataFootprint,
+      metadataFiles = metadataFiles.map(asset => asset.href).toList, // assets that are not the primary image
+      images = images,
+      thumbnails = thumbnailLinks.map(thumbnailFromLink(_, sceneId)).flatten.toList,
+      ingestLocation = None,
+      filterFields = SceneFilterFields(
+        cloudCover = None,
+        sunAzimuth = None,
+        sunElevation = None,
+        acquisitionDate = Some(feature.properties.start) // use start or end? not really sure what they should mean
+      ),
+      statusFields = SceneStatusFields(
+        thumbnailStatus = thumbnailLinks.size match {
+          case 0 => JobStatus.Queued // should kick off thumbnail creation
+          case _ => JobStatus.Success
+        },
+        boundaryStatus = JobStatus.Success, // tile and data footprints are required fields
+        ingestStatus = IngestStatus.NotIngested
+      )
+    )
+  }
+
+  protected def writeSceneToDb(scene : Scene.Create)(implicit db: DB) = {
+    Users.getUserById(systemUser) flatMap {user =>
+      user match {
+        case Some(user) =>
+          logger.info(s"\nuser: ${user.id}\ninserting scene: \n${scene}")
+          Scenes.insertScene(scene, user)
+        case _ =>
+          throw new RuntimeException("System user not found. Make sure migrations have been run, and that batch config is correct")
+      }
+    }
+  }
+
+  protected def getBandedImages(imageAssets: Seq[stac.Asset], sceneId: UUID): List[Image.Banded] = {
+    // get product
+    imageAssets.map(imageAsset =>
+      imageAsset.product match {
```

Comment:
So I think we need to be able to handle relative & absolute paths for related resources. If they are relative then we should create absolute paths that we use when creating scenes. The STAC spec at least is written to be able to support both cases and I think that's how a lot of static catalogs will be written.